src/lib/tpm.c

```diff
@@ -2538,6 +2583,16 @@ CK_RV tpm2_generate_key(
     auth->size = len;
     memcpy(auth->buffer, newauthbin, auth->size);
 
+    /*
+     * Check the ecc support of the TPM
+     */
+    if (mechanism->mechanism == CKM_EC_KEY_PAIR_GEN) {
+        rv = check_tpm_ecc(tpm->esys_ctx, tpmdat);
```

Comment:
Doesn't the RC from create let you know if the curve isn't supported? Wouldn't this be some type of specific PKCS11 error code that should be set here?

Reply:
Basically what happens is #199
You get some not nice error messages in the log, instead of gracefully telling the user his params are wrong.